src/Mod/Assembly/JointObject.py

```diff
@@ -454,51 +454,53 @@ def migrationScript(self, joint):
 
     def migrationScript2(self, joint):
         if hasattr(joint, "Object1"):
-            obj = joint.Object1[0]
-            part = joint.Part1
-            elt = joint.Object1[1][0]
-            vtx = joint.Object1[1][1]
-
-            joint.removeProperty("Object1")
-            joint.removeProperty("Part1")
-
-            # now we need to get the 'selection-root-obj' and the global path
-            rootObj, path = UtilsAssembly.getRootPath(obj, part)
-            obj = rootObj
-            elt = path + elt
-            vtx = path + vtx
-
             joint.addProperty(
                 "App::PropertyXLinkSubHidden",
                 "Reference1",
                 "Joint Connector 1",
                 QT_TRANSLATE_NOOP("App::Property", "The first reference of the joint"),
             )
+            
+            if joint.Object1 is not None:
+                obj = joint.Object1[0]
```

Comment:
trailing whitespace

Reply:
Well that's weird I tried to remove it but the commit seems empty.

Reply:
It's done.
Trailing whitespaces warnings is a waste of time IMO...

Reply:
*All* formatting discussions are a waste of time - that's why we have pre-commit.